WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/ProductFormViewModelTests.swift

```diff
@@ -103,6 +103,56 @@ final class ProductFormViewModelTests: XCTestCase {
         // Assert
         XCTAssertFalse(canShareProduct)
     }
+
+    // MARK: `canDeleteProduct`
+
+    func test_edit_product_form_with_published_status_can_delete_product() {
+        // Arrange
+        let product = MockProduct().product(name: "Test", status: .publish)
+        let viewModel = createViewModel(product: product, formType: .edit)
+
+        // Action
+        let canDeleteProduct = viewModel.canDeleteProduct()
+
+        // Assert
+        XCTAssertTrue(canDeleteProduct)
+    }
+
+    func test_add_product_form_with_published_status_cannot_delete_product() {
+        // Arrange
+        let product = MockProduct().product(name: "Test", status: .publish)
+        let viewModel = createViewModel(product: product, formType: .add)
+
+        // Action
+        let canDeleteProduct = viewModel.canDeleteProduct()
+
+        // Assert
+        XCTAssertFalse(canDeleteProduct)
+    }
+
+    func test_edit_product_form_with_non_published_status_can_delete_product() {
```

Comment:
I think the test case name conflicts with the assert 😄 (related to a previous comment about deleting product for other product statuses)

Reply:
Fixed 👍